ocrd_models/ocrd_models/ocrd_mets.py

```diff
@@ -181,8 +189,7 @@ def find_files(self, ID=None, fileGrp=None, pageId=None, mimetype=None, url=None
             # If only local resources should be returned and f is not a file path: skip the file
             if local_only and not is_local_filename(f.url):
                 continue
-            ret.append(f)
-        return ret
+            yield f
```

Comment:
Nice, just a tiny change!

So IIUC, as long as `etree.xpath()` is an iterator (and not a list), we don't need to search through all files first (and  heap them up in memory). But as soon as `pageId` is passed in, because a pageId can be covered in multiple fileIds, we are forced to enumerate the files again. Correct?

Reply:
Correct, AFAICT